apis/core/v1beta1/function_webhook.go

```diff
@@ -38,4 +43,537 @@ var _ webhook.Defaulter = &Function{}
+				if _, ok := r.Spec.Serving.Bindings[key]; ok {
+					return field.Invalid(field.NewPath("spec", "serving", "pubsub", key),
+						key,
+						"cannot use the same name as the bindings component")
+				}
+			}
+			if componentSpec.Type == "" {
+				return field.Required(field.NewPath("spec", "serving", "pubsub", key, "type"),
+					"must be specified")
+			}
+			availableTypes := []string{
+				"pubsub.kafka",
+				"pubsub.snssqs",
+				"pubsub.azure.eventhubs",
+				"pubsub.azure.servicebus",
+				"pubsub.gcp.pubsub",
+				"pubsub.hazelcast",
+				"pubsub.in-memory",
+				"pubsub.jetstream",
+				"pubsub.mqtt",
+				"pubsub.natsstreaming",
+				"pubsub.pulsar",
+				"pubsub.rabbitmq",
+				"pubsub.redis",
+			}
+			if !containsString(availableTypes, componentSpec.Type) {
+				return field.NotSupported(field.NewPath("spec", "serving", "pubsub", key, "type"),
+					componentSpec.Type,
+					availableTypes)
+			}
+		}
+	}
+
+	if r.Spec.Serving.Bindings != nil {
+		for key, componentSpec := range r.Spec.Serving.Bindings {
+			if r.Spec.Serving.Pubsub != nil {
+				if _, ok := r.Spec.Serving.Pubsub[key]; ok {
+					return field.Invalid(field.NewPath("spec", "serving", "bindings", key),
+						key,
+						"cannot use the same name as the pubsub component")
+				}
+			}
+			if componentSpec.Type == "" {
+				return field.Required(field.NewPath("spec", "serving", "bindings", key, "type"),
+					"must be specified")
+			}
+			availableTypes := []string{
+				"bindings.dingtalk.webhook",
+				"bindings.alicloud.oss",
+				"bindings.alicloud.tablestore",
+				"bindings.apns",
+				"bindings.aws.dynamodb",
+				"bindings.aws.kinesis",
+				"bindings.aws.s3",
+				"bindings.aws.ses",
+				"bindings.aws.sns",
+				"bindings.aws.sqs",
+				"bindings.azure.blobstorage",
+				"bindings.azure.cosmosdb.gremlinapi",
+				"bindings.azure.cosmosdb",
+				"bindings.azure.eventgrid",
+				"bindings.azure.eventhubs",
+				"bindings.azure.servicebusqueues",
+				"bindings.azure.signalr",
+				"bindings.azure.storagequeues",
+				"bindings.cron",
+				"bindings.gcp.pubsub",
+				"bindings.gcp.bucket",
+				"bindings.graphql",
+				"bindings.http",
+				"bindings.influx",
+				"bindings.kafka",
+				"bindings.kubernetes",
+				"bindings.localstorage",
+				"bindings.mqtt",
+				"bindings.mysql",
+				"bindings.postgres",
+				"bindings.postmark",
+				"bindings.rabbitmq",
+				"bindings.redis",
+				"bindings.rethinkdb.statechange",
+				"bindings.smtp",
+				"bindings.twilio.sendgrid",
+				"bindings.twilio.sms",
+				"bindings.twitter",
+				"bindings.zeebe.command",
+				"bindings.zeebe.jobworker",
+			}
+			if !containsString(availableTypes, componentSpec.Type) {
+				return field.NotSupported(field.NewPath("spec", "serving", "bindings", key, "type"),
+					componentSpec.Type,
+					availableTypes)
+			}
+		}
+	}
+
+	if r.Spec.Serving.Triggers != nil {
+		for index, trigger := range r.Spec.Serving.Triggers {
+			if trigger.Type == "" {
```

Comment:
I think we need to determine whether `trigger.type` is in the [scalers](https://keda.sh/docs/2.7/scalers/) set.